dagshub/common/cli.py

```diff
@@ -105,8 +114,77 @@ def upload(ctx,
 
     owner, repo_name = repo
     repo = Repo(owner=owner, name=repo_name, branch=branch)
-    repo.upload(file=filename, path=target, commit_message=message, force=update)
+    try:
+        repo.upload(file=filename, path=target, commit_message=message, force=update)
+    except RuntimeError as re:
+        # if uploaded filename already exists but update is False, a strange error message will appear
+        # since gogs tries to check last commit in this case
+        if "invalid last_commit" in str(re):
+            raise RuntimeError('Verify that the uploaded file is new') from re
+        else:
+            raise re
+
+
+@cli.command()
+@click.argument("repo", callback=validate_repo)
+@click.option("-u", "--upload-data", help="Upload data from specified url to new repository")
+@click.option("-c", "--clone", is_flag=True,  help="Clone repository locally")
+@click.option("-v", "--verbose", default=0, count=True, help="Verbosity level")
+def repo_create(
+                repo,
+                upload_data,
+                clone,
+                verbose):
+    """
+    repo_create
+    example:  dagshub repo-create sdafni.yd/yuvtutorial14 -u "http://0.0.0.0:8080/index.html" --checkout
+    """
+    logger = logging.getLogger()
+    logger.setLevel(to_log_level(verbose))
+
+    owner, repo_name = repo
+    INITIAL_DATA_FILE_NAME = 'repo_created_initial_data'
+
+    #
+    # create the repo in dagshub
+    #
+    repo = create_repo(repo_name)
+    logger.info(f"Created repo: {DEFAULT_HOST}/{owner}/{repo_name}.git")
+
+    if clone:
+        #
+        # clone locally
+        #
+        git.Git(repo_name).clone(f"{DEFAULT_HOST}/{owner}/{repo_name}.git")
+        logger.info(f"Cloned repo to folder {repo_name}")
+
+    if upload_data:
+        #
+        # get data file,
+        # if repo was cloned - copy the file to local location, commit and push to repo.
+        # if not - upload using gogs api, and delete the local copy
+        #
+        res = http_request("GET", upload_data)
+        if res.status_code != HTTPStatus.OK:
+            raise RuntimeError(f"Could not get file from source (response: {res.status_code}), repo created")
+
+        with open(INITIAL_DATA_FILE_NAME, 'w+') as fh:
```

Comment:
```suggestion
        with open(INITIAL_DATA_FILE_NAME, 'w') as fh:
```
we need to overwrite this file, otherwise the user can upload corrupted data.

Another note: the dataset can be an archive. If the user supplies an archive we need to unpack it.
It's fine if we don't support all formats, but has some ways to do [tars](https://docs.python.org/3/library/tarfile.html) and [zips](https://docs.python.org/3/library/zipfile.html), which should be enough if we warn the user about that.

The MNIST dataset is just gzipped, and archived in a specific way specifically for ML, so we don't need to do anything for the MVP, but it's still something I'd expect the client can handle

Reply:
done